x-pack/plugins/security_solution/public/management/links.ts

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { i18n } from '@kbn/i18n';
+import {
+  BLOCKLIST_PATH,
+  ENDPOINTS_PATH,
+  EVENT_FILTERS_PATH,
+  HOST_ISOLATION_EXCEPTIONS_PATH,
+  MANAGEMENT_PATH,
+  POLICIES_PATH,
+  SecurityPageName,
+  TRUSTED_APPS_PATH,
+} from '../../common/constants';
+import {
+  BLOCKLIST,
+  ENDPOINTS,
+  EVENT_FILTERS,
+  HOST_ISOLATION_EXCEPTIONS,
+  MANAGE,
+  POLICIES,
+  TRUSTED_APPLICATIONS,
+} from '../app/translations';
+import { FEATURE, LinkItem } from '../common/links/types';
+
+export const links: LinkItem = {
```

Comment:
Optional Suggestion: turn this into a function that returns `LinkItem`

__Background__
One thing that is missing today is Authz checks on these menu items. they are displayed in the side nave even if the user does not have permissions to manage Endpoint, which is inconsistent with how other Kibana side nave items work.

Reply:
these are all of the links together, they get passed through the `getDeepLinks` function which handles authz and other permissions to return the proper set of links

Reply:
Authz is not implemented for these links currently. @semd I don't know how the items you listed would help here. Endpoint management authz is determined via `useUserPrivileges().endpointPrivileges.canManageEndpoint`

Reply:
we can add that as an argument into getDeepLinks

Reply:
@paul-tavares I can't find the `canManageEndpoint` entry, did you mean `canAccessEndpointManagement`?
 
The problem with having a function is that we need all links on plugin `register` time, at that point we don't have capabilities, license, roles... and the register function needs all the deepLinks to work well at that point. So you won't have the information to filter the links in the first call.
Once the plugin is started we update the deepLinks filtering by capabilities, license, experimental flags... using the props that I described, and all pages that the user should not have access are excluded.

If your criteria to exclude a page is not possible to determine with these props (I see you check the _superuser_ role directly at some point) an alternative approach would be to keep the `export const links: LinkItem = {...}` to get the initial links, and also have an additional function `export const getFilteredLinkItem = () => {` to get the links for the update phase.

Would this work for you?

Reply:
Thanks @semd for the background here - very helpful in understanding these changes further.

#### re: the permission name

Yes, that is that I mean - sorry about that. the permission property (currently) is `canAccessEndpointManagement`.
yes, currently we check for `superuser` role, but Authz will be changing in the future as we align more closely wiht Fleet and start to allow for fine-grained access (ex. Read Only views, etc).



#### Re: determining if a menu item should be present or not

Something additional to note: Some of our menu items (`Host Isolation Exceptions` specifically) does further checks to figure out if it should be shown or not - like: is license at expected level (I assume that is what your `isPremium` property is) ++ If not at expected license level, does user have entries? (flow: license downgrades). You can check the logic here: 

https://github.com/elastic/kibana/blob/cc3412551a958631428b5d19d904bebb5426a50f/x-pack/plugins/security_solution/public/management/pages/host_isolation_exceptions/view/hooks.ts#L17-L42


That being said, having a function be available to determined the filtered set of menu items sounds like a good approach. I assume you would have a way to pass along the data we would need to figure that out (since this would be a function call, not a React hook - which is the current approach/implementation).

Reply:
Hi @paul-tavares, thanks for your feedback, and yes it makes complete sense to not assume any license level, we'll change that.
Knowing that you'll need to do some async logic to exclude links, I think we'll have to change our approach a little bit. We'll discuss that a little bit more, but by now, could you confirm that having a function with this signature:
```
export const getFilteredLinkItems = async (param: {
  enableExperimental: ExperimentalFeatures,
  licenseType: LicenseType,
  capabilities: Capabilities
}) => LinkItem[];
```
Would be enough for you?

Reply:
We need access to the Endpoint Privileges, is that inside of `capabilities`?

Reply:
These capabilities are the same as doing `useKibana().services.application.capabilities;`. 
From what I see, to get the endpoint privileges we use the `licenceType` and also the user role, which is obtained via `await security.authc.getCurrentUser();`, this is why I think the `getFilteredLinkItems` function will need to be async, and we'll probably have to pass the `services` as well.

We are going to discuss how to address this particular requirement. Probably slack will be a better place to continue this conversation.